MentorMatchmaker/MentorMatchmaker/MatchMakingDomainOperations.fs

```diff
@@ -222,38 +238,309 @@ module Matchmaking =
+                            pair.FsharpTopic.Category.CategoryName
+                            pair.CouldMentorHandleMoreWork.ToString()
+                            dumpMeetingTimes pair.MeetingTimes
+                        ]
+                ]
+                |> List.map (String.concat ";")
+                |> String.concat "\n"
+
+
+
+            let outputFileNameMatchesCsv = FileManager.saveVersionedCsv "matches" fileContents
+
+            Ok outputFileNameMatchesCsv
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+
+    let unmatchedDataDump applicantsJsonPath =
+        match FileManager.readJson<Applicants> applicantsJsonPath with
+        | Ok applicants ->
+            let fileContents = applicants |> generateUnmatchedApplicantsDumpFileContents
+
+            let outputFileName = FileManager.saveVersionedText "applicationDataDumpUnmatchedApplications" fileContents
+
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    let unmatchedPermutationsDataDump applicantsJsonPath =
+        match FileManager.readJson<Applicants> applicantsJsonPath with
+        | Ok applicants ->
+            let fileContents = applicants |> generateUnmatchedApplicantPairingPermutationsFileContents
+
+            let outputFileName = FileManager.saveVersionedText "applicationDataDumpUnmatchedApplicationsPairingPermutations" fileContents
+            
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    let generateExampleEmails matchesJsonPath = 
+        match FileManager.readJson<ConfirmedMentorshipApplication list> matchesJsonPath with
+        | Ok matches ->
+            let fileContents = matches |> EmailGenerationService.generateEmailExamplesForMatches
+
+            let outputFileName = FileManager.saveVersionedText "exampleEmailsDump" fileContents
+
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    
+    let sendEmailsMatched matchesJsonPath =
+        match FileManager.readJson<ConfirmedMentorshipApplication list> matchesJsonPath with
+        | Ok matches ->
+            // Do something
+            for pair in matches do
+                use client = createSmtpClient ()
+
+                let mails = EmailGenerationService.generateEmailsForMatch pair
+
+                // Send mentee email
+                use menteeMailMessage =
+                    new MailMessage(
+                        "mentorship@fsharp.org",
+                        pair.MatchedMentee.MenteeInformation.EmailAddress + "," + pair.MatchedMentor.MentorInformation.EmailAddress,
+                        @"FSSF Mentorship Program: Congratulations and meet your mentorship partner", // TODO generate correct title
+                        mails.MenteeEmail)
+
+                menteeMailMessage.IsBodyHtml <- true
```

Comment:
I'd create a function to create and hide the required implementation for a MailMessage.

Based on the passed info, it would either create a mentee or mentor email message.